recipes/ginkgo/all/patches/cmake-fixes.patch

```diff
@@ -0,0 +1,86 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index e75f7d6..63685f2 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -20,6 +20,12 @@ include(cmake/hip_path.cmake)
+ include(cmake/autodetect_executors.cmake)
+ include(cmake/build_type_helpers.cmake)
+ 
++if (MSVC)
++    add_compile_options(/bigobj)
++elseif (MINGW OR CYGWIN)
++    add_compile_options("-Wa,-mbig-obj")
++endif()
++
+ # Ginkgo configuration options
+ option(GINKGO_DEVEL_TOOLS "Add development tools to the build system" OFF)
+ option(GINKGO_BUILD_TESTS "Generate build files for unit tests" ON)
+diff --git a/cmake/GinkgoConfig.cmake.in b/cmake/GinkgoConfig.cmake.in
+index 0348f95..3b8be0f 100644
+--- a/cmake/GinkgoConfig.cmake.in
++++ b/cmake/GinkgoConfig.cmake.in
+@@ -129,8 +129,6 @@ set(GINKGO_OPENMP_LIBRARIES @OpenMP_CXX_LIBRARIES@)
+ set(GINKGO_OPENMP_FLAGS "@OpenMP_CXX_FLAGS@")
+ 
+ # Provide useful HIP helper functions
+-include(${CMAKE_CURRENT_LIST_DIR}/hip_helpers.cmake)
+-include(${CMAKE_CURRENT_LIST_DIR}/windows_helpers.cmake)
+ 
+ # NOTE: we do not export benchmarks, examples, tests or devel tools
+ #     so `third_party` libraries are currently unneeded.
+diff --git a/cmake/install_helpers.cmake b/cmake/install_helpers.cmake
+index ba7ea3f..7cc3ba8 100644
+--- a/cmake/install_helpers.cmake
++++ b/cmake/install_helpers.cmake
+@@ -14,16 +14,11 @@ function(ginkgo_install_library name subdir)
+         # dll is considered as runtime
+         install(TARGETS "${name}"
+             EXPORT Ginkgo
+-            LIBRARY DESTINATION ${GINKGO_INSTALL_LIBRARY_DIR}
+-            ARCHIVE DESTINATION ${GINKGO_INSTALL_LIBRARY_DIR}
+-            RUNTIME DESTINATION ${GINKGO_INSTALL_LIBRARY_DIR}
```

Comment:
What was the reason to remove these? they look correct 🤔

Reply:
Good point, that was a change made for our vcpkg port, which assumes a much higher CMake version. I'll remove it.